deliver/spec/commands_generator_spec.rb

```diff
@@ -0,0 +1,242 @@
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        app_identifier: 'abcd',
+        submit_for_review: true,
+        build_number: 'latest'
+      })
+
+      expect_run_with_options(expected_options)
+
+      Deliver::CommandsGenerator.start
+    end
+  end
+
+  describe ":init option handling" do
+    it "can use the username short flag from tool options" do
+      stub_commander_runner_args(['init', '--description', 'My description', '-u', 'me@it.com'])
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        username: 'me@it.com'
+      })
+
+      expect(Deliver::Runner).to receive(:new) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      fake_setup = "setup"
+      expect(Deliver::Setup).to receive(:new).and_return(fake_setup)
+      expect(fake_setup).to receive(:run) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      Deliver::CommandsGenerator.start
+    end
+
+    it "can use the app_identifier flag from tool options" do
+      stub_commander_runner_args(['init', '--description', 'My description', '--app_identifier', 'abcd'])
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        app_identifier: 'abcd'
+      })
+
+      expect(Deliver::Runner).to receive(:new) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      fake_setup = "setup"
+      expect(Deliver::Setup).to receive(:new).and_return(fake_setup)
+      expect(fake_setup).to receive(:run) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      Deliver::CommandsGenerator.start
+    end
+  end
+
+  describe ":generate_summary option handling" do
+    it "can use the username short flag from tool options" do
+      stub_commander_runner_args(['generate_summary', '--description', 'My description', '-u', 'me@it.com', '-f', 'true'])
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        username: 'me@it.com',
+        force: true
+      })
+
+      expect(Deliver::Runner).to receive(:new) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      fake_generate_summary = "generate_summary"
+      expect(Deliver::GenerateSummary).to receive(:new).and_return(fake_generate_summary)
+      expect(fake_generate_summary).to receive(:run) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
```

Comment:
Looks like this is the same code as the couple of tests above it. (lines 91-93, 112-114, here, and below 158-160) Do you think it would be worth while to abstract this out into its own method?